uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Query/ContactSelect.php

```diff
@@ -126,5 +150,25 @@ private function filterReturnFields() {
       $this->params['return'] = array_flip($returnFields);
     }
   }
+
+  /**
+   * Returns the conditions needed to add to the Where clause for
+   * contacts that have active leave managers
+   *
+   * @return array
+   */
+  private function activeLeaveManagerCondition() {
+    $today = date('Y-m-d');
+    $leaveApproverRelationshipTypes = $this->getLeaveApproverRelationshipsTypesForWhereIn();
+
+    $conditions = [];
+    $conditions[] = 'rt.is_active = 1';
+    $conditions[] = 'rt.id IN(' . implode(',', $leaveApproverRelationshipTypes) . ')';
+    $conditions[] = 'r.is_active = 1';
+    $conditions[] = "(r.start_date IS NULL OR r.start_date <= '$today')";
+    $conditions[] = "(r.end_date IS NULL OR r.end_date >= '$today')";
+
+    return $conditions;
+  }
```

Comment:
With this PR, this will be the third class with this same method (the others are `LeaveBalanceSelect` and `LeaveRequestSelect`). Perhaps we should see if we can refactor and extract this logic to somewhere more reusable?

Reply:
What about a `RelationshipHelper` trait having a 'getLeaveApproverCondition` method?

Reply:
It would solve the specific problem of having this method duplicated between these three classes, but we still have duplicated joins on these queries. For me, it seems that all of these are queries that support the `managed_by` param. That is, queries that return information of something managed by someone. I wonder if it would make sense to have a base `ManagedBySelect` class to centralize this logic and then these other 3 would extend it. Do you think it makes sense?

Reply:
Sounds good but not sure how complicated it's going to be and if its worth the time.